ledger-core/virtual/integration/method_test.go

```diff
@@ -532,3 +532,151 @@ func TestVirtual_CallContractFromContract_Unordered(t *testing.T) {
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB1Global)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB2Global)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB3Global)
+
+	builder := executionevent.NewRPCBuilder(gen.UniqueReference(), objectAGlobal)
+	objectAExecutionMock := runnerMock.AddExecutionMock(objectAGlobal.String())
+	objectAExecutionMock.AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB1Global, classB, "Bar", []byte{}),
+		},
+	)
+
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB2Global, classB, "Bar", []byte{}),
+		},
+	)
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB3Global, classB, "Bar", []byte{}),
+		},
+	)
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Error:  nil,
+			Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB1Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B1.Bar]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Result: requestresult.New([]byte("finish B1.Bar"), objectB1Global),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB2Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B2.Bar]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Result: requestresult.New([]byte("finish B2.Bar"), objectB2Global),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB3Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B3.Bar]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Result: requestresult.New([]byte("finish B3.Bar"), objectB3Global),
+		},
+	)
+
+	runnerMock.AddExecutionClassify(objectAGlobal.String(), flags, nil)
+	runnerMock.AddExecutionClassify(objectB1Global.String(), flags, nil)
+	runnerMock.AddExecutionClassify(objectB2Global.String(), flags, nil)
+	runnerMock.AddExecutionClassify(objectB3Global.String(), flags, nil)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallRequest.SetResend(true).ExpectedCount(3)
```

Comment:
no fields of VCallRequests are checked, I many to be checked, especially Sequence.

Reply:
done